numba/np/arraymath.py

```diff
@@ -864,6 +864,100 @@ def flat_all(a):
     return flat_all
 
 
+@register_jitable
+def _allclose_scalars(a_v, b_v, rtol=1e-05, atol=1e-08, equal_nan=False):
+    a_v_isnan = np.isnan(a_v)
+    b_v_isnan = np.isnan(b_v)
+
+    # only one of the values is NaN and the
+    # other is not.
+    if ( (not a_v_isnan and b_v_isnan) or
+            (a_v_isnan and not b_v_isnan) ):
+        return False
+
+    # either both of the values are NaN
+    # or both are numbers
+    if a_v_isnan and b_v_isnan:
+        if not equal_nan:
+            return False
+    else:
+        if np.isinf(a_v) or np.isinf(b_v):
+            return a_v == b_v
+
+        if np.abs(a_v - b_v) > atol + rtol * np.abs(b_v * 1.0):
+            return False
+
+    return True
+
+
+@overload(np.allclose)
+@overload_method(types.Array, "allclose")
+def np_allclose(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
+
+    if not type_can_asarray(a):
+        raise TypeError('The first argument "a" must be array-like')
+
+    if not type_can_asarray(b):
+        raise TypeError('The first argument "b" must be array-like')
+
+    if not isinstance(rtol, types.Float):
+        raise TypeError('The fourth argument "rtol" must be a '
+                        'floating point')
+
+    if not isinstance(atol, types.Float):
+        raise TypingError('The fifth argument "atol" must be a '
+                          'floating point')
+
+    if not isinstance(equal_nan, types.Boolean):
+        raise TypeError('The sixth argument "equal_nan" must be a '
+                        'boolean')
+
+    is_a_scalar = isinstance(a, types.Number)
+    is_b_scalar = isinstance(b, types.Number)
+
+    if is_a_scalar and is_b_scalar:
+        def np_allclose_impl_scalar_scalar(a, b, rtol=1e-05, atol=1e-08,
+                                           equal_nan=False):
+            return _allclose_scalars(a, b, rtol=rtol, atol=atol,
+                                     equal_nan=equal_nan)
+        return np_allclose_impl_scalar_scalar
+    elif is_a_scalar and not is_b_scalar:
+        def np_allclose_impl_scalar_array(a, b, rtol=1e-05, atol=1e-08,
+                                          equal_nan=False):
+            b = np.asarray(b)
+            for bv in np.nditer(b):
+                if not _allclose_scalars(a, bv.item(), rtol=rtol, atol=atol,
+                                         equal_nan=equal_nan):
+                    return False
+            return True
+        return np_allclose_impl_scalar_array
```

Comment:
You can replace this block of code by:
```suggestion
    elif is_a_scalar and not is_b_scalar:
        def np_allclose_impl_scalar_array(a, b, rtol=1e-05, atol=1e-08,
                                          equal_nan=False):
            return np.allclose(b, a, rtol=rtol, atol=atol, equal_nan=equal_nan)
        return np_allclose_impl_scalar_array
```

Reply:
I don't think this will give right results all the time, because the check depends on the ordering of `a` and `b`. For `(a, b)` the check is `abs(a - b) > atol + rtol * abs(b)`, where as for `(b, a)` the check is `abs(b - a) > atol + rtol * abs(a)`. See the RHS in both the inequalities, one uses `b` and the other uses `a`. So, `np.allclose(a, b)` might give different result from `np.allclose(b, a)` for edge cases. 

Quoting from "Notes" section of https://numpy.org/doc/stable/reference/generated/numpy.allclose.html

> The above equation is not symmetric in a and b, so that `allclose(a, b)` might be different from `allclose(b, a)` in some rare cases.

Please let me know if should still apply the above code suggestion.

Reply:
That's fine. I wasn't aware of this info.